subiquitycore/prober.py

```diff
@@ -15,45 +15,38 @@
 
 import logging
 import yaml
-from probert.network import (StoredDataObserver,
-                             UdevObserver)
+
+from probert.network import (
+    StoredDataObserver,
+    UdevObserver,
+    )
 from probert.storage import Storage
 
 log = logging.getLogger('subiquitycore.prober')
 
 
 class Prober():
     def __init__(self, opts):
-        self.opts = opts
-
-        self.probe_data = {}
         self.saved_config = None
-
-        if self.opts.machine_config:
-            with open(self.opts.machine_config) as mc:
-                self.probe_data = self.saved_config = yaml.safe_load(mc)
+        if opts.machine_config:
+            with open(opts.machine_config) as mc:
+                self.saved_config = yaml.safe_load(mc)
         log.debug('Prober() init finished, data:{}'.format(self.saved_config))
 
     def probe_network(self, receiver):
-        if self.opts.machine_config:
+        if self.saved_config is not None:
             observer = StoredDataObserver(
                 self.saved_config['network'], receiver)
         else:
             observer = UdevObserver(receiver)
         return observer, observer.start()
 
     def get_storage(self, probe_types=None):
```

Comment:
Hrm, I do wish we called this probe_storage() to match probe_network().  But that's not a big deal.

Reply:
One day...